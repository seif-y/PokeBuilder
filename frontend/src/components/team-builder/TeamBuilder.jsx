import React, { useState } from "react";
import TeamMember from "./team-selector/TeamMember";
import styles from "./TeamBuilder.module.css";

export default function TeamBuilder() {
    const [team, setTeam] = useState(Array(6));

    function saveTeam() {
        console.log(team);
    }

    function updateTeam(index, teamMember) {
        let teamMembers = [...team];
        teamMembers[index] = teamMember;
        setTeam(teamMembers);
    }

    function renderTeamSlots() {
        const teamMemberSlots = Array(6);
        for (let i = 0; i < 6; i++) {
            teamMemberSlots[i] = (
                <TeamMember key={`teamSlot${i}`} index={i} onUpdate={(index, obj) => updateTeam(index, obj)} />
            );
        }
        return teamMemberSlots;
    }

    return (
        <div id="modal-root" className={styles.teamBuilderContainer}>
            <div className={styles.titleContainer}>
                <h1> Team Builder </h1>
                <button className={styles.saveButton} onClick={() => saveTeam()}>
                    Save
                </button>
            </div>
            <input className={styles.teamNameInput} placeholder="Team Name" />
            <div className={styles.teamContainer}>{renderTeamSlots()}</div>
            <textarea className={styles.descriptionInput} placeholder="Team description" />
        </div>
    );
}
