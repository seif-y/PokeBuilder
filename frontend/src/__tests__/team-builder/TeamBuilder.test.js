import React from "react";
import axios from "axios";
import { render, screen } from "@testing-library/react";
import renderer from "react-test-renderer";
import TeamBuilder from "../../components/team-builder/TeamBuilder";
import userEvent from "@testing-library/user-event";
import { MemoryRouter } from "react-router";

jest.mock("axios");

describe("General TeamBuilder Tests", () => {
    beforeEach(() => {
        render(
            <MemoryRouter>
                <TeamBuilder />
            </MemoryRouter>
        );
    });

    test("TeamBuilder contains team name input", () => {
        const teamNameInput = screen.getByPlaceholderText("Team Name");
        expect(teamNameInput).toBeDefined();
    });

    test("TeamBuilder contains team description text area", () => {
        const descriptionTextArea = screen.getByPlaceholderText("Team description");
        expect(descriptionTextArea).toBeDefined();
    });

    test("Save button does POST on click", async () => {
        const saveButton = screen.getByText("Save");
        expect(saveButton).toBeDefined();
        axios.post.mockImplementation(() => Promise.resolve({ data: { _id: "" } }));
        userEvent.click(saveButton);
        expect(axios.post).toHaveBeenCalledWith("/api/teams", expect.any(Object), expect.any(Object));
    });
});

test("TeamBuilder snapshot test", () => {
    const tree = renderer.create(<TeamBuilder />).toJSON();
    expect(tree).toMatchSnapshot();
});
