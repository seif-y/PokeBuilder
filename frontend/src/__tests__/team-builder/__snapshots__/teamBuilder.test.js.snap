// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TeamBuilder snapshot test 1`] = `
<div
  className="teamBuilderContainer"
>
  <div
    className="topBar"
  >
    <span
      className="title"
    >
      TEAM BUILDER
    </span>
    <input
      className="teamNameInput"
      onChange={[Function]}
      placeholder="Team name"
    />
    <div
      className="hSpacer"
    />
    <div
      className="button white"
      onClick={[Function]}
    >
      Save
    </div>
  </div>
  <div
    className="pageContent"
  >
    <div
      className="vSpacer"
    />
    <div
      className="teamContainer"
    >
      <div
        className="container"
        onClick={[Function]}
      >
        <div
          className="addIconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </div>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <div
          className="addIconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </div>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <div
          className="addIconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </div>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <div
          className="addIconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </div>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <div
          className="addIconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </div>
      </div>
      <div
        className="container"
        onClick={[Function]}
      >
        <div
          className="addIconContainer"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-colorAction MuiSvgIcon-fontSizeLarge"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M19 13h-6v6h-2v-6H5v-2h6V5h2v6h6v2z"
            />
          </svg>
        </div>
      </div>
    </div>
    <div
      className="vSpacer"
    />
    <textarea
      className="textArea descriptionInput"
      onChange={[Function]}
      onClick={[Function]}
      placeholder="Team description"
    />
    <div
      className="vSpacer"
    />
  </div>
</div>
`;
